src/orient/rational-calc.orient

```diff
@@ -0,0 +1,550 @@
+      posts_per_sector_per_year = eposts_per_sector_per_year + fposts_per_sector_per_year
+
+      // TODO: model single ticket vs multiple tickets
+      ElectionPoSt:
+        eposts_per_sector_per_year = blocks_in_a_year / filecoin_sectors
+
+      FallbackPoSt:
+        fposts_per_sector_per_year = (secs_in_year / fallback_period) * fallback_ratio
+
+  Mining:
+    ElectionPoSt:
+      describe(epost_challenged_sectors, "Number of challenged sectors during Election PoSt")
+      describe(epost_challenged_sectors_fraction, "Fraction of challenged sectors during Election PoSt")
+      describe(epost_data_access, "Time to access data for Election PoSt", secs)
+      describe(epost_data_access_parallel, "Time to access data for Election PoSt in parallel", secs)
+      describe(epost_time, "Time to generate an Election PoSt", secs)
+      describe(epost_time_parallel, "Time to generate an Election PoSt in parallel", secs)
+      describe(epost_tickets_gen, "Time to generate all the tickets in Election PoSt", secs)
+      describe(epost_tickets_gen_parallel, "Time to generate all the tickets in Election PoSt in parallel", secs)
+
+      epost_challenged_sectors = rig_sectors_min * epost_challenged_sectors_fraction
+      epost_data_access = post_data_access * epost_challenged_sectors
+      epost_data_access_parallel = epost_data_access_parallel / rig_storage_parallelization  
+      epost_tickets_gen = (post_inclusions_time + post_ticket_gen) * epost_challenged_sectors
+
+      epost_time = epost_data_access + epost_tickets_gen
+      epost_time_parallel = epost_data_access_parallel + epost_tickets_gen_parallel
+
+Rationality:
+  describe(breakeven_nodes, "number of nodes / sector / challenge period after which it's cheaper to store the entire sector, than re-encode those")
+  describe(polling_time, "Time between two PoSt proofs", secs)
+  describe(challenge_periods_per_month, "Number of challenge periods in a month")
+  describe(regeneration_fraction, "Fraction of nodes of a sector for which re-encoding them every `polling_time` costs as much as storing them")
+
+  breakeven_nodes = ((cost_gib_per_month *  sector_size_gib) / (cost_node_encoding * challenge_periods_per_month))
+  challenge_periods_per_month = secs_in_month / polling_time
+  regeneration_fraction = breakeven_nodes / nodes
+
+  Costs:
+    Compute:
+      describe(cost_node_encoding, "Cost of encoding a single node")
+      describe(nodes_per_second, "Nodes encoded per second using the fastest hardware")
+      describe(nodes_per_year, "Nodes encoded per year using the fastest hardware")
+      describe(hash_gb_per_second, "GB that can be hashed per second")
+      describe(hash_gib_per_second, "GiB that can be hashed per second")
+
+      hash_gib_per_second = hash_gb_per_second * to_i
+      nodes_per_second = hash_gib_per_second / kdf_hash_size_gib
+      nodes_per_year = nodes_per_second * secs_in_year 
+      cost_node_encoding = rig_malicious_cost_per_year / nodes_per_year 
```

Comment:
This does not include the cost of transferring memory and cost of hashing or does it ? Asking because the latest supranational spreadsheet doesn't use this model of computing the price. Maybe it's hidden in this model but it doesn't appear like it.